sync.py

```diff
@@ -70,30 +114,59 @@ def all_messages(credentials, full_sync=False) -> int:
             .execute()
         )
 
-        messages = results.get("messages", [])
+        messages_from_api = results.get("messages", [])
+        if not messages_from_api:
+            print("No new messages found in this page.")
+            break # No messages in this page, could be end of sync for non-full syncs
 
-        total_messages += len(messages)
-        for i, m in enumerate(messages, start=total_messages - len(messages) + 1):
+        for i, m_summary in enumerate(messages_from_api):
+            current_total_count = total_messages_processed_in_run + current_batch_count + 1
```

Comment:
There is currently a [multi thread PR](#5) open, that does batching. I would like to use this approach for speeding up things.